### PR TITLE
[3.x] add show function

### PR DESCRIPTION
**Pull request by sanbuphy**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

https://github.com/open-mmlab/mmdetection/issues/9312

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmdet/apis/inference.py [hunk 1]
```diff
@@ -201,3 +204,64 @@ async def async_inference_detector(model, imgs):
...
+    """
+    if hasattr(model, 'module'):
+        model = model.module
+    if isinstance(img, str):
+        image = imread(img)
+    else:
+        image = img.copy()
+
+    # init visualizer
+    visualizer = DetLocalVisualizer(
```
Comment (hhaAndroid) [inline mmdet/apis/inference.py:250]: This is not universal and cannot be used with other visualizers.

Comment (sanbuphy) [reply mmdet/apis/inference.py:250]: How can I  modify it better, or Is there any way to visualizer the results more simpler APi than visualizer.

Comment (hhaAndroid) [inline mmdet/apis/inference.py:250]: visualizer = VISUALIZERS.build(model.cfg.visualizer)

File: mmdet/apis/inference.py [hunk 2]
```diff
@@ -201,3 +204,64 @@ async def async_inference_detector(model, imgs):
...
+def show_result_pyplot(model: BaseDetector,
+                       img: Union[str, np.ndarray],
+                       result: DetDataSample,
+                       opacity: float = 0.5,
+                       title: str = '',
+                       draw_gt: bool = True,
+                       draw_pred: bool = True,
+                       wait_time: float = 0,
+                       show: bool = True,
+                       save_dir=None,
```
Comment (hhaAndroid) [inline mmdet/apis/inference.py:218]: typehint

Comment (sanbuphy) [reply mmdet/apis/inference.py]: OK,The problem has been solved.

File: mmdet/apis/inference.py [hunk 3]
```diff
@@ -202,3 +207,54 @@ async def async_inference_detector(model, imgs):
...
+            that means "forever". Defaults to 0.
+        out_file (str, optional): Path to output file. Default to None.
+    """
+    if hasattr(model, 'module'):
+        model = model.module
+    if isinstance(img, str):
+        image = imread(img)
+    else:
+        image = img.copy()
+    try:
```
Comment (hhaAndroid) [inline mmdet/apis/inference.py:241]: VISUALIZERS.get_instance()

File: mmdet/apis/inference.py [hunk 4]
```diff
@@ -202,3 +207,54 @@ async def async_inference_detector(model, imgs):
     torch.set_grad_enabled(False)
     results = await model.aforward_test(data, rescale=True)
     return results
+
+
+def show_result_pyplot(model: BaseDetector,
```
Comment (hhaAndroid) [inline mmdet/apis/inference.py:212]:
```suggestion
def show_result(model: BaseDetector,
```

File: mmdet/apis/inference.py [hunk 5]
```diff
@@ -202,3 +207,54 @@ async def async_inference_detector(model, imgs):
...
+        name=title,
+        image=image,
+        data_sample=result,
+        draw_gt=draw_gt,
+        draw_pred=draw_pred,
+        wait_time=wait_time,
+        out_file=out_file,
+        show=True)
+
+    vis_img = visualizer.get_image()
```
Comment (hhaAndroid) [inline mmdet/apis/inference.py:258]: The following code does not require

File: mmdet/apis/inference.py (hunk 4 above)
Comment (hhaAndroid) [reply mmdet/apis/inference.py:212]: What I'm thinking is whether this function should only support pypolt? If your purpose is to only support pypolt, then you really don't need to consider general

Comment (hhaAndroid) [reply mmdet/apis/inference.py:212]: Maybe you can take a look first https://github.com/open-mmlab/mmdetection/pull/8975/files

**Pull request closed without merging**